tests/integration/conftest.py

```diff
@@ -152,20 +154,31 @@ async def run_cairo_test_runner(
     return run_cairo_test_runner
 
 
-@pytest.fixture(name="protostar_project_root_path", scope="module")
-def protostar_project_root_path_fixture(tmp_path_factory: TempPathFactory) -> Path:
-    tmp_path = tmp_path_factory.mktemp("data")
-    return tmp_path / "tmp_project"
+class CreateProtostarProjectFixture(Protocol):
+    def __call__(self) -> ContextManager[ProtostarFixture]:
+        ...
 
 
-@pytest.fixture(name="protostar", scope="module")
-def protostar_fixture(
+@pytest.fixture(name="create_protostar_project", scope="module")
+def create_protostar_project_fixture(
     session_mocker: MockerFixture,
-    protostar_project_root_path: Path,
+    tmp_path_factory: TempPathFactory,
     signing_credentials: Tuple[str, str],
-) -> ProtostarFixture:
-    return build_protostar_fixture(
-        mocker=session_mocker,
-        project_root_path=protostar_project_root_path,
-        signing_credentials=signing_credentials,
-    )
+):
+    @contextmanager
```

Comment:
does this have to be context manager? From what i know, we're using i.e. devnet in similar manner and it doesn't have to be used with `with` keyword

Reply:
Unlike the devnet fixture, the side effect that needs to be reverted is caused when the function is called. There's no point of cleaning up if the function is never called.

Reply:
got it